openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/UserRepository.java

```diff
@@ -126,8 +126,13 @@ public User getByEmail(UriInfo uriInfo, String email, Fields fields) {
     if (userString == null) {
       throw EntityNotFoundException.byMessage(CatalogExceptionMessage.entityNotFound(USER, email));
     }
-    return withHref(
-        uriInfo, setFieldsInternal(JsonUtils.readValue(userString, User.class), fields));
+    User user = JsonUtils.readValue(userString, User.class);
+    setFieldsInternal(user, fields);
+    setInheritedFields(user, fields);
+    // Clone the entity
+    User entityClone = JsonUtils.deepCopy(user, User.class);
```

Comment:
@mohityadav766 whats the need for this?

Reply:
to get entity by Email in case of UserRepository